### ISSUE TITLE
Unable to connect to server with error message on HL2

### ISSUE THREAD

**Issue by Futo23** [MicrosoftDocs/mixed-reality#2212]
After completing all the steps in the tutorial and deploying to HoloLens 2, I started the app. After playing the game on Unity Editor, the following message was displayed on HoloLens 2.

> Failed to connect to server after testing each known IP. 

The development environment is as follows.

- Unity 2019.2.21f1
- Visual Studio 2019 ( Version 16.5.3 )

If there are any items that require additional settings or solutions, would you please let me know?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: c749260f-d284-74f9-ffff-9afd2069ef81
* Version Independent ID: 04aff1f3-9c67-07f5-4fe2-6b1fd256e9f3
* Content: [Multi-user capabilities tutorials - 3. Connecting multiple users - Mixed Reality](https://docs.microsoft.com/en-us/windows/mixed-reality/mrlearning-sharing(photon)-ch2)
* Content Source: [mixed-reality-docs/mrlearning-sharing(photon)-ch2.md](https://github.com/MicrosoftDocs/mixed-reality/blob/master/mixed-reality-docs/mrlearning-sharing(photon)-ch2.md)
* Product: **mixed-reality**
* Technology: **mr-dev**
* GitHub Login: @jessemcculloch
* Microsoft Alias: **jemccull**

**Comment by martinwilter** [comment]
Hi @Futo23,

Sounds like Photon is trying to connect to a local server. Do you have 'Use Name Server' checked as shown in this [image](https://docs.microsoft.com/en-us/windows/mixed-reality/images/mrlearning-sharing/tutorial1-section7-step1-3.png) from the [Connecting the Unity project to the PUN application](https://docs.microsoft.com/en-us/windows/mixed-reality/mrlearning-sharing(photon)-ch1#connecting-the-unity-project-to-the-pun-application) section?

When you press Play in Unity, do you see the following message displayed in the Game window?

>PhotonLobby.OnJoinRoom()
>Current room name: RoomNNNN
>Other players in room: 0
>Total players in room: 1

**Comment by Futo23** [comment]
Hi @martinwilter,

Thank you for your quick reply. Yes, I have 'Use Name Sever' checked.
The settings are as shown in the photo below.

![image](https://user-images.githubusercontent.com/16267066/81972588-c8033600-965d-11ea-807b-de2f7abc3dac.png)

When I press the play button in Unity Editor, the above message is displayed.

![image](https://user-images.githubusercontent.com/16267066/81973023-7b6c2a80-965e-11ea-808b-5992c5f97e7b.png)

**Comment by martinwilter** [comment]
@Futo23, thanks for confirming that Photon works as expected in Unity, which should indicate that PUN is set up correctly.

Can you double-check that you have the **InternetClientServer** and **PrivateNetworkClientServer** capabilties enabled? 

If those are enabled and you still reive the error, we suggest that you try to create a mobile hotspot and connect your HoloLens 2 to it instead of your local Wi-Fi to rule out any local network issues.

You can also try to set Fixed Region in your PhotonServerSettings and see if that helps. For region list, you can refere to Photon's [Regions](https://doc.photonengine.com/en-us/realtime/current/connection-and-authentication/regions) documentation.

We just re-tested the tutorial project on our HoloLens 2, and everything worked as expected. Our test point was tag [multi-user-capabilities-v2.3.0.0](https://github.com/microsoft/MixedRealityLearning/tree/multi-user-capabilities-v2.3.0.0) so you could try to clone the repo, check out that tag, add your credentials, and see if that project works on your HL2.

Hope this helps. Let us know how it goes.

**Comment by Futo23** [comment]
@martinwilter , Thank you for your kindness.

I have tried all the solutions that you have told me, but the tutorial didn't work.

Below is a summary of what I tried.
・[Check] Capabilities : I have InternetClientServer and PrivateNetworkClientServer checked.
・[Check] Mobile Hot Spot : Connect to mobile hotspot on iPhone and launch app ( didn't work )
・[Change] Set fix Region in my PhotonSeverSettings
・[Change] Cloned tutorial, checked out tags ( multi-user-capabilities-v2.3.0.0 ) and add Unity's PUN Setup, launch application

**Comment by martinwilter** [comment]
@Futo23, sorry to hear these steps didn't help, was the error the same "Failed to connect to server after testing each known IP"?

Attached is a link to a zip archive with the AppX we tested when you first reported the issue last week, it was built from the [multi-user-capabilities-v2.3.0.0](https://github.com/microsoft/MixedRealityLearning/tree/multi-user-capabilities-v2.3.0.0) tag. Can you install it on your HoloLens and see you are able to connect to Photon using that app?

Zip archive name: _MRTK Tutorials - Sharing v2.3.0.0_1.0.0.0_ARM_Test.zip
App display name on HoloLens: _MRTK Tutorials - Sharing v2.3.0.0

https://netorg4603737-my.sharepoint.com/:u:/g/personal/team_onginnovations_com/EVdlyrdcLBdBsNc0C4VHWHEBrfFVXSK0_AJL-EaMVGPbpw?e=ogWGON

**Comment by Futo23** [comment]
@martinwilter , Thank you for your support.

Yes, I got the same error when I tried the project for that tag.
> "Failed to connect to server after testing each known IP"?

I have deployed the appX file you sent to HL2. Then I started the application and confirmed the operation normally. 

I would like to try the tutorial again from scratch. Thank you for your polite response.

**Comment by martinwilter** [comment]
Hi @Futo23, thanks for the update, so the appx file we built worked as expected on your HoloLens?

**Comment by Futo23** [comment]
Hi @martinwilter, Yes, it worked as expected.

**Comment by martinwilter** [comment]
Thanks for confirming, @Futo23.

I built that appx from the [multi-user-capabilities-v2.3.0.0](https://github.com/microsoft/MixedRealityLearning/tree/multi-user-capabilities-v2.3.0.0) tag and you also deployed the project at that tag. Given that, it seems likely that the issue is with Visual Studio and not Unity.

Is your PC running the latest Windows 10 and do you have Windows SDK build 18362 or later installed?

I'm including my Visual Studio installation configuration file, [vsconfig.zip](https://github.com/MicrosoftDocs/mixed-reality/files/4684128/vsconfig.zip), which should reflect my Visual Studio installation when I built the appx that worked on your HoloLens 2. You can export yours and compare it to mine, or simply import mine. See the [Import or export installation configurations](https://docs.microsoft.com/en-us/visualstudio/install/import-export-installation-configurations?view=vs-2019) documentation for instructions on how to export/import the installation configuration file.

**Comment by MMMMysticY** [comment]
Hi @martinwilter  
I‘m sorry and I also get the same issue.  I follow the tutorial and when it runs in unity everything works fine
![222](https://user-images.githubusercontent.com/35412654/82968903-e889a380-a000-11ea-974f-9663e02408ba.png)

but when it runs in my HL2, it shows that "Failed to connect to server after testing each known IP."  I also tried the appx "MRTK Tutorials - Sharing v2.3.0.0_1.0.0.0_ARM_Test.zip" when i use it, nothing in the scene, no errors, only a CPU window. Does this indicate that the program is running normally on my HL2?

![111](https://user-images.githubusercontent.com/35412654/82969307-effd7c80-a001-11ea-92bb-577e967e57d5.png)

thanks for your help

**Comment by martinwilter** [comment]
@MMMMysticY

>I also tried the appx "MRTK Tutorials - Sharing v2.3.0.0_1.0.0.0_ARM_Test.zip" when i use it, nothing in the scene, no errors, only a CPU window. Does this indicate that the program is running normally on my HL2?

You should see the same as when you run the project in Unity, so if you re-launch the app, **_MRTK Tutorials - Sharing v2.3.0.0**, you should see the message:

>PhotonLobby.OnJoinRoom()
>Current room name: RoomNNNN
>Other players in room: 0
>Total players in room: 1

and the Rocket Launcher experience.

Your PUN settings looks fine. 

PS. It might not be a good idea to share your personal account IDs or keys publicly. Feel free to remove the image from your post above.

**Comment by Futo23** [comment]
Hi @martinwilter , Thank you for your explaination.

My pc is running the Windows 10 Pro ( Version 2004, OS Build 19041.207 ) 
and I have Windows SDK 10.0.18362.0 / 10.0.19041.0 .

I compared the difference between your VSConfig and my VSConfig.
However, I didn't find any difference, and all were the same.

My Visual Studio build settings are:

<img src="https://user-images.githubusercontent.com/16267066/83051583-2df1b380-a089-11ea-92d7-09e8927294f0.png" width="300"> 

<img src="https://user-images.githubusercontent.com/16267066/83051654-4530a100-a089-11ea-9ea6-3ca394c3b567.png" width="300"> 

I changed the target SDK version to 10.0.18362.0 but I still got the same error.

I am very grateful for reaching out to solve difficult problems.

**Comment by martinwilter** [comment]
Thanks for sharing that, @Futo23. We will try to reproduce the issue. Would you mind sharing your vsconfig file with us?

>※ I changed the target SDK version to 10.0.18362.0 but I still got the same error.

Did you change the Target Platform Version or the Minimum version to 10.0.18362.0? I have my Target Platform Version set to 10.0.18362.0. If you haven't already, would you mind trying with Target Platform Version set to 10.0.18362.0?  If that still doesn't work, maybe try to remove 10.0.19041.0 from your Visual Studio.

I saw in the Photon forum discussion you shared that someone was able to resolve their issue and that he/she had the Target Platform Version set to 10.0.18362.0.

**Comment by MMMMysticY** [comment]
Hi @martinwilter  Sorry for not answering you for a long time. I have tried to change the target SDK version to 10.0.18362.0 and now it works as expected. Thank you very much! I really appreciate your help!

**Comment by martinwilter** [comment]
@MMMMysticY, glad to hear that resolved the issue, thanks for letting us know!

@Futo23, when you have a chance, please let us know if this (Target Platform Version set to 10.0.18362.0) resolved the issue for you as well.

**Comment by Futo23** [comment]
Hi @martinwilter, 
The application worked as expected. I am very grateful for your support for a long time.

It was caused by that Target SDK Version was set to Latest Installed (10.1009041.0) on the Unity's Build Settings.

Above settings caused errors related .dll file  when building with Visual Studio 2019.
Although I changed the Target SDK Version 10.0.18362.0 on Visual Studio, I got same errors.

I solved the error with the following steps:
**Changed Target SDK Version  from Latest Installed to 10.0.18362.0 on Unity's Build Settings**

This issue can be closed.
